### PR TITLE
07-MemoryManagementKernelModule: add module implementation

### PR DESCRIPTION
**Pull request by prosolvo4th**
Signed-off-by: yuriy.podgorniy <yuriy.pidgorniy@gmail.com>

### CODE REVIEW

File: 07-MemoryManagementKernelModule/allocation.c [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+			if (deltaFree[i].tv_sec != 0)
+				deltaFreeNS[i] += 1000000000;
+		}
+
+		bubbleSort(deltaAllocNS, attempts);
+		bubbleSort(deltaFreeNS, attempts);
+
+		pr_info("vmalloc:2^%02d+%d	%09ld	%09ld	%09ld	%09ld	%09ld	%09ld",
+			power, oversize,
+			deltaAllocNS[1], deltaAllocNS[attempts/2],
```
Comment (yekovalyov) [inline 07-MemoryManagementKernelModule/allocation.c]: Please consider the difference between average and median value. Is it require to sort the obtained results?

File: 07-MemoryManagementKernelModule/allocation.c [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+			deltaFree[i] = sub(t2, t1);
+		}
+
+		long deltaAllocNS[attempts], deltaFreeNS[attempts];
+
+		i = 0;
+		for (; i < attempts; i++) {
+			deltaAllocNS[i] = deltaAlloc[i].tv_nsec;
+			if (deltaAlloc[i].tv_sec != 0)
+				deltaAllocNS[i] += 1000000000;
```
Comment (yekovalyov) [inline 07-MemoryManagementKernelModule/allocation.c]: Is only possible value of tv_sec is 1?

Comment (prosolvo4th) [reply 07-MemoryManagementKernelModule/allocation.c]: No, its not.

**Comment by prosolvo4th** [comment]
Changes: use min(), max(), use averaging instead median, fix bug "deltaAlloc[i].tv_sec > 1".
Update example log

File: 07-MemoryManagementKernelModule/allocation.c [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+		avrAllocNS /= attempts;
+		avrFreeNS /= attempts;
+
+		pr_info("kzalloc:2^%02d+%d	%09ld	%09ld	%09ld	%09ld	%09ld	%09ld",
+			power, oversize, minAllocNS, avrAllocNS, maxAllocNS,
+			minFreeNS, avrFreeNS, maxFreeNS);
+
+		oversize++;
+		oversize %= 2;
```
Comment (yekovalyov) [inline 07-MemoryManagementKernelModule/allocation.c:173]: In my opinion one line for oversize update is enough. Like oversize = 1 - oversize
or oversize ^= 1

**Pull request merged**